### DISCUSSION TITLE
I'd like to know if it's possible to support 6GB GPUs

### DISCUSSION THREAD

**Discussion by sbrooke** [Chia-Network/bladebit#360] [General]
Tried Alpha 1 anyway. Failed with a cuda memory allocation error. There are many GTX 1060 6GB cards out there from former ETH mining operations. The requirement to buy a newer GPU with 8GB to plug into a system with 256GB ram minimum is counter to the conservation aspect of Chia. MMX supports 6GB cards. Let's get Bladebit to the same.

**Comment by sbrooke** [comment]
Looking at this again and the "GPU RAM required" says only 6.135GB. My GTX 1060 6GB comes back with 5.86GB free. So close! Hoping it's not that hard to trim down the required memory just a little.

**Comment by harold-b** [comment]
The required vRAM should be around 5.6 or something like that. However, it is aligned to system page size boundary, which could add up if you have large pages. I wonder if that's your case.

Either way, I believe some users tested on a 6G card and it would not allow them to allocate the ~5.6G anyway. You can try an experimental thing (although I added it, I am not officially supporting it currently), when compiling.

Change [this line](https://github.com/Chia-Network/bladebit/blob/cuda-compression/cuda/CudaPlotConfig.h#L8) to say `256u` instead of `128u` and it should lower the memory requirements. I am not sure if it will still be enough for the driver to allow the allocation on the 6G card, but you can give it a try.

Of course, you have to compile it yourself to test it with that change.

**Comment by sbrooke** [comment]
Thanks! Someone else suggested the same thing. Although I see the instructions for building it, I’m not familiar with downloading the code to begin with. Git download either isn’t working or I’m doing it wrong. The latter is quite likely. If someone can point me in the right direction I’ll be happy to test.

Also, the error message is mixing GB and GiB. I don’t know if that accounts for some of the perceived difference in required vram.

**Comment by sbrooke** [comment]
Never mind, I think I might have figured it out. Testing in a few, I hope…

**Comment by sbrooke** [comment]
As far as I can tell I've built the right version. If I do "bladebit help" there are references to cudaplot for the type and the compression level option. However, when I run it I get "unexpected argument 'cudaplot'". I'll keep trying different things.

**Comment by harold-b** [comment]
It's possible the wrong one was built. The command should be something like:

```bash
git clone https://github.com/Chia-Network/bladebit.git
cd bladebit
git checkout cuda-compression
mkdir -p build-release && cd build-release
cmake .. && cmake --build . --config Release --target bladebit_cuda -j$(nproc --all)
```
Then you should see the `bladebit_cuda` binary.


FYI, a user on keybase tried the above solution with `256u` and it worked with a 6G card.

**Comment by sbrooke** [comment]
`-- Build files have been written to: /home/plot/bladebit/build-release
gmake: *** No rule to make target 'bladebit_cuda'.  Stop.
`
I'll keep poking at it

**Reply by Zealot88** [reply]
Hi @sbrooke, how did you resolve this? Trying to get a GTX 980 Ti to work (I know it's older than the 1060, but should be faster) I have searched for how to fix this but didn't find a solution.

**Comment by sbrooke** [comment]
I keep trying different driver configurations and can't get it to work. Best I could do was:

`nvcc fatal   : Unsupported gpu architecture "'compute_89'"`

**Comment by graemes** [comment]
You need to build with Cuda 12 (compute_89 is for Ada support).

If not possible just comment out the '-gencode=arch=compute_89,code=sm_89 # Ada' line in CMakeLists.txt

**Comment by sbrooke** [comment]
Switched to Rocky 9 and it built without issue, once I figured out all of the dependencies.

GTX 1060 6GB (hobbled on a 8x slot, no less) benchmarks at 11.3 minutes. This really needs to be incorporated into the mainline.

**Comment by sbrooke** [comment]
Tried some compression levels. c3 and c5 both create the plot in about 10 minutes and at about 90GB in size. No significant difference between the two.

**Comment by sbrooke** [comment]
FYI, I’ve been running 6.3 minute C05 plots on the GTX 1060 6gb for months. Works great. Wish I didn’t have to jump through hoops to compile it.
